pulp_ostree/app/tasks/stages.py

```diff
@@ -88,32 +96,31 @@ async def run(self):
         """Create relations between each OSTree object specified in DeclarativeContent objects."""
         async for batch in self.batches():
             updated_commits = []
-            updated_refs = []
+            updated_objs = []
             for dc in batch:
                 if dc.extra_data.get("parent_commit"):
                     updated_commits.append(self.associate_parent_commit(dc))
-                elif dc.extra_data.get("head_commit"):
-                    updated_refs.append(self.associate_head_commit(dc))
+                elif dc.extra_data.get("obj_commit"):
+                    updated_objs.append(self.associate_obj_commit(dc))
 
-            OstreeCommit.objects.bulk_update(
+            await sync_to_async(OstreeCommit.objects.bulk_update)(
```

Comment:
Ah, no here is another one (async fix).
On a related note: We should really not update content at any time. In a future PR we should use the content futures to resolve these associations in the first stage before saving the content.

Reply:
Well, we are not updating the content at any time. We are doing the very same thing as we do in pulp_container (https://github.com/pulp/pulp_container/blob/ad9c8816d412410d2e2d2994c4ea853758036035/pulp_container/app/tasks/sync_stages.py#L155-L158 -> refs_commits; https://github.com/pulp/pulp_container/blob/ad9c8816d412410d2e2d2994c4ea853758036035/pulp_container/app/tasks/sync_stages.py#L473-L475 -> objs_commits) or not?